pkg/v2/crud/traverse.go

```diff
@@ -25,9 +60,12 @@ func primaryOrFirstTraverse(property prop.Property, query *expr.Expression, call
 }
 
 type traverser struct {
-	nav             prop.Navigator                 // stateful navigator for the resource being traversed
-	callback        func(nav prop.Navigator) error // callback function to be invoked when target is reached
-	elementStrategy elementStrategy                // strategy to select element properties to traverse for multiValued properties
+	addByEqFilter        bool                   // true if valueByEqFilter should be updated by the Eq filter
+	valueByEqFilter      interface{}            // value to be updated using the Eq filter
```

Comment:
this feels bolted on to me - the traverser is generic and we're adding very specific functionality. is there a way to implement this via the elementStrategy and/or callback instead?

Reply:
Defined a single callback taking a value as a parameter, ignored in most cases
Removed eq-filter - specific attributes from the traverse struct
if value is passed - it considered as the one to be updated once the eq filter is parsed - this way looks more generic and scalable to me.